sshmux.go

```diff
@@ -401,6 +403,47 @@ func sshmuxServer(configFile string) {
 	}
 }
 
+func sshmuxServer(configFile string) {
+	configFileBytes, err := os.ReadFile(configFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = json.Unmarshal(configFileBytes, &config)
+	if err != nil {
+		log.Fatal(err)
+	}
+	sshConfig := &ssh.ServerConfig{
+		ServerVersion:           "SSH-2.0-taokystrong",
+		PublicKeyAuthAlgorithms: ssh.DefaultPubKeyAuthAlgos(),
+	}
+	for _, keyFile := range config.HostKeys {
+		bytes, err := os.ReadFile(keyFile)
+		if err != nil {
+			log.Fatal(err)
+		}
+		key, err := ssh.ParsePrivateKey(bytes)
+		if err != nil {
+			log.Fatal(err)
+		}
+		sshConfig.AddHostKey(key)
+	}
+	if config.Address == config.ProxiedAddress {
+		if config.Address == "" {
+			log.Println("No address specified, defaulting to 0.0.0.0:8022")
+			go sshmuxListenAddr("0.0.0.0:8022", sshConfig, false, false)
```

Comment:
Now that all of these listeners run in their own goroutines (which they did not), I'm afraid the program will terminate as soon as `main()` returns. Has this been integration-tested?